### ISSUE TITLE
Combobox ControlValueAccessor should always reflect the value from the input

### ISSUE THREAD

**Issue by gergo-papp** [ni/nimble#1168]
## 😯 Problem to Solve

Currently `nimble-combobox` emits a `Symbol('not found')` object when the user enters a string input which doesn't correspond to any existing combobox option. 

## 💁 Proposed Solution(s)

We want to change the combobox to behave just like an autocompleting text input, where whatever the user types (or autocompletes / selects from the dropdown) is emitted as a string.

This is mirrored in AzDO. [User Story 2810741](https://dev.azure.com/ni/DevCentral/_workitems/edit/2810741): Change Nimble combobox CVA to reflect value of input field

**Comment by gergo-papp** [comment]
Also it would be nice to support arbitrary values while using the combobox in a formControl

**Comment by jattasNI** [comment]
@atmgrifter00 is our combobox guru so he's the best person to offer guidance on this. (He's OOO today so likely won't respond until next week).

**Comment by atmgrifter00** [comment]
Hey @gergo-papp, [we already have an issue](https://github.com/ni/nimble/issues/682) that is similar to this one. What are your thoughts on the 2 proposed solutions there (which I think are close to what you suggest)? I don't have a strong opinion between the two options I outlined.

**Comment by robert-bob-1** [comment]
Hello @atmgrifter00,
The problem we are facing is: we can't programatically set the combobox render value to something that is not in its list of options (it is set to empty). From what I can tell the solutions provided by Gergo and in the issue you opened don't fix this (correct me if I'm wrong).
Could the solution be adding a new boolean input `customValues`? If it is set we bypass checking if the model value appears in the list, and instead of printing `OPTION_NOT_FOUND` we print the custom value. Would this be clean enough or is there another possible solution?

This first modification fixes the issue of `OPTION_NOT_FOUND` output. 
![image](https://github.com/ni/nimble/assets/90975931/e2283de4-9742-4e2c-a0e5-c7e38bcc373d)

And this fixes the problem of setting the combobox rendered value to empty instead of the custom value
![image](https://github.com/ni/nimble/assets/90975931/28d60bd2-a19f-44b1-8e85-07c052554b10)

**Comment by atmgrifter00** [comment]
Hey @robert-bob-1, another direction we're considering going (contingent on a new feature request that I have yet to file), is removing the model association behavior from the Combobox altogether. The "model" value would simply always be the text. For cases we've seen where a user wants an actual model associated with a selectable entity, that component does _not_ support arbitrary values. To enable a filerable dropdown component, we intend to create a feature request to add that capability to the `Select`.

**Comment by robert-bob-1** [comment]
In that case I think we will leave it as it is, and update our component in the future when this feature is implemented.

**Comment by rbell517** [comment]
Just to reinforce the need for this, we another case of wanting arbitrary values in a combobox in Test Insights and we've worked around it by checking the value for `OPTION_NOT_FOUND` and instead getting the value directly from the underlying control.

**Comment by fredvisser** [comment]
@atmgrifter00 - Did https://github.com/ni/nimble/commit/501e91b66bc55a27c9acd7958fb3ab1545353dc0 fix this issue?

**Comment by atmgrifter00** [comment]
> @atmgrifter00 - Did [501e91b](https://github.com/ni/nimble/commit/501e91b66bc55a27c9acd7958fb3ab1545353dc0) fix this issue?

Sort of? The _real_ fix is ultimately to change the CVA of the `Combobox` to only support text as the model value.

**Comment by rajsite** [comment]
> Sort of? The _real_ fix is ultimately to change the CVA of the `Combobox` to only support text as the model value.

The [refresh issue](https://github.com/ni/nimble/issues/1303) was linking to this open issue under the label ` Control Value Accessor behavior` so we can either create a new issue to represent that more succinctly or continue using this issue for that.

**Comment by jattasNI** [comment]
I changed the title and description to make this specifically about the CVA.

**Comment by MuthuRajan-S** [comment]
This issue is discussed in today's UI working group meeting. Due to this [limitation](https://nimble.ni.dev/storybook/index.html?path=/docs/components-combobox--docs#angular), several UI workflows in Stratus enterprise app has to use a work-around to accommodate for arbitrary string values than provided list of values by referencing combo-box from html using `NimbleComboboxDirective` or `ElementRef`.

The nimble-select documentation clearly stated `Use nimble-combobox if users should be able to enter a custom value in addition to selecting from the list.`

<img width="783" height="196" alt="Image" src="https://github.com/user-attachments/assets/0cbd9ce9-621c-4971-b815-c4191b7baa6b" />

But unfortunately, combo-box didn't implicit support this behavior. There are several solutions suggested in discussions in this issue about changing the CVA and there is an active work item created for this issue : [User Story 2810741](https://dev.azure.com/ni/DevCentral/_workitems/edit/2810741): Change Nimble combobox CVA to reflect value of input field.

So unearthing this nimble issue (hidden for a long while), in order to get people's attention, start working on it and get rid of work-around codes.

FYI @jattasNI , @fredvisser , @rajsite , @hemkumarrao-ni , @gokuldharan-ni

**Comment by MuthuRajan-S** [comment]
> Need to do a survey of all the uses of combobox in the product and see if they can be switched to a select, now that the select supports filtering. Once we've done that, we can potentially change the behavior of the combobox. [@MuthuRajan-S](https://github.com/MuthuRajan-S) if this is something you have bandwidth for in the near term, feel free. Otherwise we will have to get to it sometime in the future.

@m-akinc , We don't think our squad has enough bandwidth to work on this right now. We have to get to it sometime in the future.
